File: src/common/updates/UpdateRAM.java
package common.updates;

import common.GameContainer;
import common.util.Logger;
import de.hsh.daniel.RAM;

public class UpdateRAM {

    public static void update(RAM game, Object arg, GameContainer gameContainer) {
        Logger.log( UpdateRAM.class + ": " + game, arg );
       if(arg instanceof String) {
           String message = (String) arg;
            switch (message) {
                case UpdateCodes.DefaultCodes.exitToMainGUI:
                    gameContainer.showMainMenu();
                    System.gc();                        //remind the garbage collector. he may trow some unused objects away after the game session should be closed.
                    break;
            }
        }
    }
}

File: Games/GameCollection/src/main/common/updates/UpdateRAM.kt
package common.updates

import main.common.GameContainer
import common.codes.EngineObserverCodes
import common.logger.Logger
import de.hsh.daniel.RAM

object UpdateRAM {
    fun update(game: RAM,
               arg: Any?,
               gameContainer: GameContainer) {
        Logger.log(UpdateRAM::class.java.toString() + ": " + game,
                   arg)
        if (arg is String) {
            if (EngineObserverCodes.exitToMainGUI == arg) {
                gameContainer.showMainMenu()
                System.gc() //remind the garbage collector. he may trow some unused objects away after
                // the game session should be closed.
            }
        }
    }
}
